raiden/network/transport/matrix.py

```diff
@@ -536,22 +531,55 @@ def _network_name(self) -> str:
     def _private_rooms(self) -> bool:
         return bool(self._config.get('private_rooms'))
 
-    def _login_or_register(self):
+    def _login_or_register(self, prev_user_id=None, prev_access_token=None):
+        base_username = to_normalized_address(self._raiden_service.address)
+        _match_user = re.match(
+            f'^@{re.escape(base_username)}.*:{re.escape(self._server_name)}$',
+            prev_user_id or '',
+        )
+        if _match_user:  # same user as before
+            self.log.debug('Trying previous user login', user_id=prev_user_id)
+            self._client.set_access_token(user_id=prev_user_id, token=prev_access_token)
+
+            # test previous user credentials with quick /sync call
```

Comment:
Please update this comment as well (or just remove it, the log message in the except case makes it pretty clear what's happening here)